### PR TITLE
Move network technologies to their own projects

### PR DESCRIPTION
**Pull request by fhunleth**
This formalizes the separation between the core VintageNet code and the network
interface-specific code for handling WiFi, wired Ethernet, USB gadget code,
etc.

The primary reason for doing this is establish the practice of extending
VintageNet for network technologies that are not appropriate or practical to
maintain in the core.

A side benefit includes not pulling in dependencies for unneeded technologies.

### CODE REVIEW

**Comment by fhunleth** [comment]
I intended this to be a draft. I'd like to improve the docs more before merging.

File: README.md [hunk 1]
```diff
@@ -157,9 +157,9 @@ The only required key in the configuration maps is `:type`. All other keys
 follow from the type. `:type` should be set to a module that implements the
 `VintageNet.Technology` behaviour. The following are included:
 
-* `VintageNet.Technology.Ethernet` - Standard wired Ethernet
-* `VintageNet.Technology.WiFi` - Client configurations for 802.11 WiFi
-* `VintageNet.Technology.Mobile` - Cellular configurations (likely to be
+* `VintageNetEthernet` - Standard wired Ethernet
+* `VintageNetWiFi` - Client configurations for 802.11 WiFi
+* `VintageNetMobile` - Cellular configurations (likely to be
```
Comment (jjcarstens) [inline README.md]: I think this is no longer "likely" and we can remove that bit in the () (assuming we also move it to `VintageNetMobile`

File: README.md [hunk 2]
```diff
@@ -92,8 +92,8 @@ domain if using WiFi. In your main `config.exs`, add the following:
 config :vintage_net,
   regulatory_domain: "US",
   config: [
-    {"eth0", %{type: VintageNet.Technology.Ethernet, ipv4: %{method: :dhcp}}},
-    {"wlan0", %{type: VintageNet.Technology.WiFi}}
+    {"eth0", %{type: VintageNetEthernet, ipv4: %{method: :dhcp}}},
+    {"wlan0", %{type: VintageNetWiFi}}
```
Comment (jjcarstens) [inline README.md]: My 2¢ here is that we actually move the`Network interface configuration` below to their respective libs (i.e. `vintage_net_ethernet`). And here we explain that to configure the needed interface, you need the dep. Then a quick "Heres how the config would look", and reference more details to their lib pages

File: lib/vintage_net/interface.ex [hunk 3]
```diff
@@ -99,15 +99,27 @@ defmodule VintageNet.Interface do
...
   defp technology_from_config(_missing),
     do:
       raise(ArgumentError, """
       Missing :type field.
 
-      See the help for VintageNet.Technology.Ethernet and VintageNet.Technology.WiFi for
-      example configurations.
+      This should be set to a network technology. These are provided in other libraries. For example,
+      for WiFi, add `vintage_net_wifi` as a dependency and set this to `VintageNetWifi`.  See docs
+      and cookbook for example configurations.
```
Comment (jjcarstens) [inline lib/vintage_net/interface.ex]: Do we have a link for this? I think including the link could be beneficial here

File: README.md (hunk 2 above)
Comment (fhunleth) [reply README.md]: Yes. Completely agree on this comment and the other doc comments - just didn't have time yet.

File: README.md (hunk 1 above)
Comment (fhunleth) [reply README.md]: I deleted `VintageNetMobile` to stop teasing people.

File: lib/vintage_net/interface.ex (hunk 3 above)
Comment (fhunleth) [reply lib/vintage_net/interface.ex]: I simplified this by pointing generically to the docs. Perhaps when this is published a good URL will be available.

**Comment by fhunleth** [comment]
@jjcarstens Alright, I made another pass on the docs. The main change was to move technology-specific docs to their respective repos.

**Pull request merged**